### PR TITLE
 PLAT-64255: Optimize ExpandableItem render performance

### PR DESCRIPTION
**Pull request by germboy**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
`ExpandableItem` performs an additional update/render after mounting due to its use of `Transition` - which performs an idle job in order to render its children. This behavior is generally ideal in that it optimizes mount time initially. However when used in views (`Panel`), it can block animations and result in longer transition time between panels.

### Resolution
We can utilize `ExpandableSpotlightDecorator` to only mount the transition container when it becomes focused (to prepare for the assumed next open state) or when specifically toggled open. When performing this via a job, we can prevent this transition container rendering when simply moving a pointer over & out of an `ExpandableItem` - where we weren't expecting to open the expandable - saving us the trouble of needlessly rendering.

### CODE REVIEW

**Comment by germboy** [comment]
I meant to open this as a draft PR (if it even matters) - but I'd like to discuss whether we think this is the right general approach for the perf mounting issue.

I notice that tests are failing with this change - although it's not immediately clear to me why. But I can clean that up as we progress this ticket.

**Comment by germboy** [comment]
Due to the nature of handling child container rendering in `ExpandableSpotlightDecorator`, it will be necessary to use and test against the full version of our expandable components, rather than the base components when testing contained children. As far as I know, we're expecting developers to use the full component so I don't believe this is a breaking change, but correct me if I'm wrong.
That being the case, I've updated the tests to reflect this behavior. In the case of `DatePicker` and `TimePicker`, the full component also uses i18n, which alters some of the expected results.

**Comment by ryanjduffy** [comment]
It was a bit weird that we were `mount()`-ing base components anyways so not too concerned about the test changes.

File: packages/moonstone/ExpandableItem/ExpandableItem.js [hunk 1]
```diff
@@ -250,6 +250,15 @@ const ExpandableItemBase = kind({
...
 		open: PropTypes.bool,
 
+		/**
+		 * Prevents rendering the transition container on mount.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @private
+		 */
+		preventTransitionContainerRender: PropTypes.bool,
```
Comment (ryanjduffy) [inline packages/moonstone/ExpandableItem/ExpandableItem.js]: This prop name is a bit unconventional in the framework. It's very specific to the implementation details of the effect. What are some other options we should consider that are a bit more generalized? Maybe `hideChildren` as is used in Panel?

File: packages/moonstone/ExpandableItem/ExpandableItem.js [hunk 2]
```diff
@@ -250,6 +250,15 @@ const ExpandableItemBase = kind({
 		 */
 		open: PropTypes.bool,
 
+		/**
+		 * Prevents rendering the transition container on mount.
```
Comment (ryanjduffy) [inline packages/moonstone/ExpandableItem/ExpandableItem.js]: Independent of the prop naming comment, the "on mount" portion isn't really accurate in this context.

**Comment by ryanjduffy** [comment]
Code looks sound to me. I won't have time to test in detail before EOD so I'm going to defer merging for now.

**Pull request merged**